docs/userguides/development.md

```diff
@@ -7,22 +7,73 @@ In this guide, we are going to show you more details on how to build an applicat
+
+2. Create a `bots/` folder in the root directory of your project. Then develop bots in this folder as separate scripts (Do not include a __init__.py file).
+
+3. Create a `bot/` folder with a `__init__.py` file that will include the instantiation of your `SilverbackApp()` object.
+
+The `silverback` cli automatically searches for python scripts to run as bots in specific locations relative to the root of your project.
+It will also be able to detect the scripts inside your `bots/` directory and let you run those by name (in case you have multiple bots in your project).
+`silverback` can also deal with a `bot/` folder with an `__init__.py` file that is defined as the runner.
+
+```{note}
+It is suggested that you create the instance of your `SilverbackApp()` object by naming the variable `bot`, since `silverback` will autodetect that variable name when loading your script file.
+```
+
+Another way you can structure your bot is to create a `bot` folder and define a runner inside of that folder as `__init__.py`.
+
+If you have a more complicated project that requires multiple bots, naming each bot their own individual name is okay to do, and we encourage you to locate them under the `bots/` folder relative to the root of your project.
+This will work fairly seamless with the rest of the examples shown in this guide.
+
+If your bot's module name is `example.py` (for example), you can run it like this:
+
+```bash
+silverback run example --network your:network:of:choice
+```
+
+If the variable that you call the `SilverbackApp()` object is something other than `bot`, you can specific that by adding `:{variable-name}`:
+
+```bash
+silverback run example:{variable-name} --network your:network:of:choice
+```
+
+We will automatically detect all scripts under the `bots/` folder automatically, but if your bot resides in a location other than `bots/` then you can use this to run it:
+
+```bash
+silverback run folder.example:app --network your:network:of:choice
+```
+
+With a `bot/__init__.py` setup, you can run with:
+
+```bash
+silverback run bot --network your:network:of:choice
+```
+
+```{note}
+It is suggested that you develop your modules as scripts to keep your deployments simple. If you have a deep understanding of containerization, and distributed applications, you can set your bots up however you'd like, and then create your own container definitions for deployments. For a simple deployment ecosystem, develop your bots as scripts, and avoid developing your applications as packages. (Do not include an __init__.py module, and do not use python modules for reusable code). If you follow this simple architecture, your deployments will be almost entirely hands off.
+```
+
 ## Creating an Application
 
 Creating a Silverback Application is easy, to do so initialize the `silverback.SilverbackApp` class:
 
 ```py
 from silverback import SilverbackApp
 
-app = SilverbackApp()
+bot = SilverbackApp()
```

Comment:
Question: what do we think about renaming `SilverbackApp` to `SilverbackBot`, to match the rest of this?

Reply:
seems like a major change, no? Like should bump the major for that, idk

Reply:
This PR is already a minor bump, I was planning for v0.6.0